rasa/core/agent.py

```diff
@@ -392,156 +312,110 @@ class Agent:
-        except ModelNotFound as e:
-            raise ModelNotFound(
-                f"You are trying to load a model from '{model_path}', "
-                f"which is not possible. \n"
-                f"The model path should be a 'tar.gz' file or a directory "
-                f"containing the various model files in the sub-directories "
-                f"'core' and 'nlu'. \n\n"
-                f"If you want to load training data instead of a model, use "
-                f"`agent.load_data(...)` instead. {e}"
-            )
-
-        core_model, nlu_model = get_model_subdirectories(model_path)
-
-        if not interpreter and nlu_model:
-            interpreter = rasa.core.interpreter.create_interpreter(nlu_model)
-
-        domain = None
-        ensemble = None
-
-        if core_model:
-            domain = Domain.load(os.path.join(core_model, DEFAULT_DOMAIN_PATH))
-            ensemble = (
-                PolicyEnsemble.load(
-                    core_model,
-                    new_config=new_config,
-                    finetuning_epoch_fraction=finetuning_epoch_fraction,
-                )
-                if core_model
-                else None
-            )
-
-            # ensures the domain hasn't changed between test and train
-            domain.compare_with_specification(core_model)
+        model_metadata, graph_runner = cls.unpack_model(model_path)
 
-        return cls(
-            domain=domain,
-            policies=ensemble,
-            interpreter=interpreter,
+        agent = cls(
+            domain=model_metadata.domain,
             generator=generator,
             tracker_store=tracker_store,
             lock_store=lock_store,
             action_endpoint=action_endpoint,
-            model_directory=model_path,
             model_server=model_server,
             remote_storage=remote_storage,
-            path_to_model_archive=path_to_model_archive,
+            graph_runner=graph_runner,
+            model_path=model_path,
+            model_id=model_metadata.model_id,
         )
 
-    def is_core_ready(self) -> bool:
-        """Check if all necessary components and policies are ready to use the agent."""
-        return self.is_ready() and self.policy_ensemble is not None
+        agent.initialize_processor()
+        return agent
 
-    def is_ready(self) -> bool:
-        """Check if all necessary components are instantiated to use agent.
-
-        Policies might not be available, if this is an NLU only agent."""
+    @staticmethod
+    def unpack_model(
+        model_path: Union[Text, Path]
+    ) -> Tuple[ModelMetadata, GraphRunner]:
+        """Unpacks a model from a given path using the graph model loader."""
+        model_tar = get_latest_model(model_path)
+        if not model_tar:
+            raise ModelNotFound(f"No model found at path {model_path}.")
+
+        tmp_model_path = tempfile.mkdtemp()
+        return loader.load_predict_graph_runner(
```

Comment:
Since everything is eager, once the the graph runner is loaded, we no longer need disk -> we can drop the `tmp_model_path`, right? In turn means we can use the context manager of `tempfile` here

Reply:
Otherwise we're accumulating unpacked dirs 😬